frontend/components/common/AppProviders.tsx

```diff
@@ -9,10 +9,17 @@ import {
    QueryClient,
    QueryClientProvider,
 } from '@tanstack/react-query';
+import dynamic from 'next/dynamic';
 import * as React from 'react';
-import { AlgoliaProps, InstantSearchProvider } from './InstantSearchProvider';
+import type {
+   AlgoliaProps,
+   InstantSearchProviderProps,
+} from './InstantSearchProvider';
 
 const customTheme = extendTheme(theme);
+const InstantSearchProvider = dynamic<InstantSearchProviderProps>(() =>
+   import('./InstantSearchProvider').then((mod) => mod.InstantSearchProvider)
```

Comment:
It seems like an import at run time will result in a longer time before the asset beings downloadin. *This file* has to be downloaded and ran before `./InstantSearchProvider` is even requested.

I'd be more inclined to have an additional component (an optional wrapper) here (outside `AppProviders` or inside it) that only includes `InstantSearchProvider`.

Actually.. since `<InstantSearchProvider>` is really simple, let's not put it in a new component, but merely use it directly on the pages that need it. Thoughts?

Reply:
Actually bundles (also dynamically linked ones) are linked via script tag in the page html at SSR time. 
So the client immediately sees the entire list of chunks required by a specific page.
There shouldn't be the need to parse the code and then decide if an additional js chunk needs to be downloaded or not

Reply:
> There shouldn't be the need to parse the code and then decide if an additional js chunk needs to be downloaded or not

Maybe I'm missing something, but if a chunk isn't conditionally downloaded, then what makes the import "dynamic"? And how would making it dynamic and conditionally including (as you've done here) reduce the size of the downloaded JS?

Reply:
Next.js dynamic import are some sort of code splitting hints.
At compile time dynamically imported dependencies are bundled as separate chunks.
At runtime, the imported module is evaluated on the server side. 
If the dynamic component is conditionally loaded based on a falsey condition, it won't be included in the initial bundle.
Otherwise if the condition in truthy, it is included along with the initial server-side rendered HTML.